### DISCUSSION TITLE
How to make the position of SplitLayout update with the customized value?

### DISCUSSION THREAD

**Discussion by Seikaijyu** [AllenDang/giu#535] [Q&A] [answered]
I want to update the split position of splitlayout. Is this possible

**Comment by Seikaijyu** [comment, marked as answer]
And can I not drag the separator bar manually?

**Reply by AllenDang** [reply]
In giu, `Child` is a panel container, SplitLayout is made of two `Child` with a splitter.
You could play with `Child`, `Row` a little bit to understand what are they and how to combine them together to implement complex layout.

To change the split position, notice there is a variable which controls it, try to change it and see what will happen.

**Reply by Seikaijyu** [reply]
Cannot change the split position externally?

**Reply by gucio321** [reply]
@SuperLiveYou couldn't you just update sashPos argument to SplitLayouot? It is a pointer so updating it should probably work

**Reply by Seikaijyu** [reply]
But when I use it, the parameter does not need a pointer. @gucio321

**Reply by Seikaijyu** [reply]
That's why I ask this

**Reply by Seikaijyu** [reply]
I try to define a variable and update it, but it's useless

**Reply by Seikaijyu** [reply]
I try to define a variable and update it, but it's useless

**Reply by gucio321** [reply]
Are you at master version of code? Iirc there is already a pointer...

**Reply by gucio321** [reply]
![Screenshot_20220719-163143.png](https://user-images.githubusercontent.com/73652197/179776273-671852ee-2dc6-4301-962c-c96ddfde28ff.png)

**Reply by Seikaijyu** [reply]
Go automatically installed it for me, I'm not sure

**Reply by Seikaijyu** [reply]
@gucio321

**Reply by Seikaijyu** [reply]
I think I'm sure my version is the latest. Is your version too old? @gucio321

**Reply by Seikaijyu** [reply]
Oh, I just looked at it. Maybe it's my version. I'll try to modify it after work

**Reply by gucio321** [reply]
`go get -u github.com/AllenDang/giu@master` or instead branch name the commit hash

**Reply by Seikaijyu** [reply]
Thanks, the problem has been solved
